### PR TITLE
respect include header option

### PR DESCRIPTION
**Pull request by abist**
This PR fixes https://github.com/microsoft/azuredatastudio/issues/13217

### CODE REVIEW

File: src/sql/workbench/contrib/query/browser/gridPanel.ts
Comment (alanrenmsft) [inline src/sql/workbench/contrib/query/browser/gridPanel.ts:393]: did you mean to use the configuration mentioned in the issue 'queryEditor.results.copyIncludeHeaders'?

Comment (kburtram) [reply src/sql/workbench/contrib/query/browser/gridPanel.ts:393]: I think this setting used in the PR is only for exporting results to CSV.  The copy behavior would likely be a different setting (though honestly I got the design around some of these query editor settings).

Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/query/browser/gridPanel.ts:393]: You should query this value in the copy callbacks to ensure that you have the correct value - it currently looks like if someone runs a query and then changes the value of this setting the grid will still have the original value if they then copy the results.

File: src/sql/workbench/contrib/query/browser/gridPanel.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/query/browser/gridPanel.ts:393]: You should query this value in the copy callbacks to ensure that you have the correct value - it currently looks like if someone runs a query and then changes the value of this setting the grid will still have the original value if they then copy the results.

Comment (abist) [reply src/sql/workbench/contrib/query/browser/gridPanel.ts]: Changed the implementation to check the setting during copying, and verified the setting is respected in real time.

File: src/sql/workbench/contrib/editData/browser/gridActions.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/editData/browser/gridActions.ts:34]: Any reason you're not using the dependency injection for this?

Comment (abist) [reply src/sql/workbench/contrib/editData/browser/gridActions.ts:34]: Just following previous practices (dataService)

**Review by alanrenmsft — APPROVED** [review]

File: src/sql/workbench/contrib/editData/browser/gridActions.ts
Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/editData/browser/gridActions.ts:34]: Data service isn't an actual service that can be injected so that's probably why that wasn't done there. For DI-able services we should use the DI annotations.

File: src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:109]: You shouldn't need to pass in the configuration service anymore - just move the param to the end of the list and createInstance will inject it for you.

File: src/sql/workbench/contrib/editData/browser/gridActions.ts
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/editData/browser/gridActions.ts:50]: Would suggest creating these with instantiation service instead to avoid needing to pass in the config service directly

File: src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts
Comment (Charles-Gagnon) [reply src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts:109]: (once you update it to also use DI)

**Comment by alanrenmsft** [comment]
@abist, are you still working on this one?

**Comment by abist** [comment]
@alanrenmsft It should be mergeable, but I didn't make the change Charles was suggesting. I can either merge it now and use instantiation service later, or merge this later if it's not urgent because I'm working on other things right now

**Review by Charles-Gagnon — APPROVED** [review]
I went and fixed them, once checks pass go ahead and merge this in

**Pull request merged**
